index.ts

```diff
@@ -735,20 +738,23 @@ export function createProxyCache<
       }
 
       // Otherwise try to get from non-memory cache
-      if (!options.cacheOutsideMemory?.messages || !options.getItem) return;
+      if (options.cacheOutsideMemory?.messages && options.getItem) {
+        const stored = await options.getItem<T["message"]>("message", messageID);
+        if (stored && options.cacheInMemory?.messages) {
+          bot.cache.messages.memory.set(messageID, stored);
+        }
 
-      const stored = await options.getItem<T["message"]>("message", messageID);
-      if (stored && options.cacheInMemory?.messages) {
-        bot.cache.messages.memory.set(messageID, stored);
+        if (stored) return stored;
       }
 
-      if (stored) return stored;
-
-      if (options.fetchIfMissing?.messages && channelId && guildId) {
-        return fetchers.fetchMessage(messageID, BigInt(guildId), BigInt(channelId));
+      const channelID = channelId ?? bot.cache.messages.channelIDs.get(BigInt(id));
+      if (options.fetchIfMissing?.messages && channelID) {
+        const guildID = guildId ?? channelID ? bot.cache.channels.guildIDs.get(BigInt(channelID)) : undefined;
```

Comment:
`guildId ?? channelID`?

Reply:
> `guildId ?? channelID`?

check the whole line
` const guildID = guildId ?? (channelID ? bot.cache.channels.guildIDs.get(BigInt(channelID)) : undefined);`

Reply:
Somehow didn't see that